packages/myst-cli/src/init/jupyter-book/syntax.ts

```diff
@@ -0,0 +1,279 @@
+        if (maybeNewLines !== undefined) {
+          // Backup existing file
+          await fs.rename(path, backupFilePath);
+
+          // Write modified result
+          session.log.info(chalk.dim(`Backed up original version of ${path} to ${backupFilePath}`));
+          await fs.writeFile(path, maybeNewLines.join('\n'));
+        }
+      }
+      break;
+    case '.ipynb': {
+      // Upgrade each cell of the notebook
+      const data = (await fs.readFile(path)).toString();
+      const notebook = JSON.parse(data) as INotebookContent;
+      const cellDidUpgrade = await Promise.all(
+        notebook.cells
+          .filter((cell) => cell.cell_type === 'markdown')
+          .map(async (cell) => {
+            // Try and upgrade the cell
+            const maybeNewLines = await upgradeContent(
+              (cell.source as string[])
+                // Strip newlines
+                .map((line) => line.replace('\n', '')),
+            );
+            // Did we compute new state?
+            if (maybeNewLines !== undefined) {
+              // Write to cell and indicate modification
+              cell.source = maybeNewLines.map((line) => `${line}\n`);
+              return true;
+            } else {
+              return false;
+            }
+          }),
+      );
+
+      // Do we need to update the notebook?
+      if (cellDidUpgrade.some((x) => x)) {
+        // Backup existing file
+        await fs.rename(path, backupFilePath);
+
+        // Write modified result
+        const newData = JSON.stringify(notebook);
+        session.log.info(chalk.dim(`Backed up original version of ${path} to ${backupFilePath}`));
+        await fs.writeFile(path, newData);
+      }
+    }
+  }
+}
+export async function upgradeContent(documentLines: string[]): Promise<string[] | undefined> {
+  let didUpgrade = false;
+
+  for (const transform of [upgradeGlossary, upgradeNotes]) {
+    const nextLines = await transform(documentLines);
+    didUpgrade = didUpgrade || nextLines !== undefined;
+    documentLines = nextLines ?? documentLines;
+  }
+
+  return didUpgrade ? documentLines : undefined;
+}
+
+const admonitionPattern =
+  /^(attention|caution|danger|error|important|hint|note|seealso|tip|warning|\.callout-note|\.callout-warning|\.callout-important|\.callout-tip|\.callout-caution)$/;
+
+async function upgradeNotes(documentLines: string[]): Promise<string[] | undefined> {
+  const data = documentLines.join('\n');
+  const mdast = mystParse(data);
+
+  const caseInsenstivePattern = new RegExp(admonitionPattern.source, admonitionPattern.flags + 'i');
+  const directiveNodes = selectAll('mystDirective', mdast);
+  const mixedCaseAdmonitions = directiveNodes.filter((item) => {
+    const name = (item as any).name as string;
+    return name.match(caseInsenstivePattern) && !name.match(admonitionPattern);
+  });
+  mixedCaseAdmonitions.forEach((node) => {
+    const start = node.position!.start.line;
```

Comment:
Is this `!` assertion reasonable?

Reply:
I think that it should be fine in this case, I think in general an early return of:

```
if (!node.position) return
```

Might be safer in all cases. If this works for you though (e.g. in the turing way), then it should be fine.

Reply:
I think it's safe as we're using a non-transformed pass, and we're looking for `mystDirective` nodes which are from the parse phase.

Reply:
Awesome. Will look at this and merge today or tomorrow!